docs/design/direct-assign-volume.md

```diff
@@ -0,0 +1,139 @@
+at the root of the volume on the host. The contents of this file will be defined as follows:
+
+```go
+type DiskMountInfo struct {
+        // Device: source device for this volume
+        Device string json:"device,omitempty"
+
+        // VolumeType: type associated with this volume (ie, block)
+        VolumeType string json:"volumeType,omitempty"
+
+        // TargetPath: path which this device should be mounted within the guest
+        TargetPath string json:"targetPath,omitempty"
+
+        // FsType: filesystem that needs to be used to mount the storage inside the VM
+        FsType string json:"fsType,omitempty"
+
+        // Options: additional options that might be needed to mount the storage filesystem
+        Options string json:options,omitempty"
+}
+```
+
+The block device will be formatted, but not mounted on the host, by the CSI node driver. Through `csiPlugin.json`,
+the runtime will identify that the existing mount should not be mounted into the guest by way of the shared filesystem,
+but instead the described device should be attached to the VM, and volume mount information should be provided to the
+agent for mounting the in-guest block device to the TargetPath in the guest for consumption by the container(s) in
+the pod.
+
+## Managing the mount
+
+Normally any node agent, including the CSI node driver, would be able to directly check the status of the mountpath, 
+and look at usage statistics. When the volume mount is not mounted on the host, this is no longer feasible. We need
+to introduce a mechanism for gathering similar information on nodes that are utilizing Kata Containers with
+direct-assigned volumes.
+
+These actions will be achieved by interacting with a host binary, `kata-runtime`, which will in turn interact with
+the applicable shim-kata-v2 instance on the host which will interact with the applicable in-guest kata-agent.
+
+### kata-runtime changes:
+
+The host agent (CSI node driver, as an example) will call into `kata-runtime`. kata-runtime behaves as the API for sandbox-level management interactions. In this case, we propose augmenting `kata-runtime` to include a filesystem, fs, command as follows:
+
+```bash
+kata-runtime fs —help
+NAME:
+kata-runtime fs [subcommand] [options]— interact with mounted filesystem volume
+
+SUBCOMMANDS:
+ - get-stats <pod-id> <host-blk-device-name> — show status and usage details for mounted filesystem in pod <pod-id> backed by the given block device, <host-blk-device-name>.
+ - resize <pod-id> <host-blk-device-name> <size> — update the mounted volume to <size> for the mounted filesystem in pod <pod-id> backed by the given block device, <host-blk-device-name>.
+ ```
+
+Based on the sandbox id provided, the Kata-runtime handler will then interact with the appropriate sandbox to
+either gather the filesystem stats for the provided block-device backed volume, or to facilitate resizing the
+mounted volume.
+
+### containerd-shim-kata-v2 changes:
```

Comment:
See comments above about specifying file names in headings. Here I'd say "Kata Containers Runtime changes" or something similar since it's not 100% clear to all that `containerd-shim-kata-v2` is a program, is in fact a Kata one and not a Containerd one, and is actually the Kata 2.x runtime ;)